sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateFormatter.scala

```diff
@@ -77,14 +96,14 @@ trait LegacyDateFormatter extends DateFormatter {
 
 class LegacyFastDateFormatter(pattern: String, locale: Locale) extends LegacyDateFormatter {
   @transient
-  private lazy val fdf = FastDateFormat.getInstance(pattern, locale)
```

Comment:
is it needed to remove `lazy`?

Reply:
needed https://github.com/apache/spark/pull/28592/files#diff-7f589e01d3e5e5ea284c1622527d4984R52 here, to fail if invaild